### ISSUE TITLE
使用m1 mac运行示例的lenet模型的代码时，卡住不动,但是在linux可以正常运行

### ISSUE THREAD

**Issue by which520** [PaddlePaddle/Paddle#50912]
### bug描述 Describe the Bug

<img width="1010" alt="image" src="https://user-images.githubusercontent.com/44831650/221366179-b509e358-6af5-4812-8eec-a3236fa469d9.png">

**Comment by 2742195759** [comment]
这个行为确实是不符合预期的，如果在linux下可以正常运行，那么mac下卡死应该是bug了。同学可以粘贴一下可以复现的示例代码吗，我本地验证一下。

**Comment by which520** [comment]
import paddle
import numpy as np
from paddle.vision.transforms import Normalize

transform = Normalize(mean=[127.5], std=[127.5], data_format='CHW')

train_dataset = paddle.vision.datasets.MNIST(mode='train', transform=transform)
test_dataset = paddle.vision.datasets.MNIST(mode='test', transform=transform)


lenet = paddle.vision.models.LeNet(num_classes=10)
model = paddle.Model(lenet)


model.prepare(paddle.optimizer.Adam(parameters=model.parameters()), 
              paddle.nn.CrossEntropyLoss(),
              paddle.metric.Accuracy())


model.fit(train_dataset, epochs=5, batch_size=64, verbose=1)
model.evaluate(test_dataset, batch_size=64, verbose=1)

model.save('./output/mnist')
model.load('output/mnist')

img, label = test_dataset[0]
img_batch = np.expand_dims(img.astype('float32'), axis=0)


out = model.predict_batch(img_batch)[0]
pred_label = out.argmax()
print('true label: {}, pred label: {}'.format(label[0], pred_label))
from matplotlib import pyplot as plt
plt.imshow(img[0])

**Comment by SigureMo** [comment]
@which520 我这边 M1 mac、Paddle 版本为 2.4.1 和 develop 都无法复现

<img width="804" alt="image" src="https://user-images.githubusercontent.com/38436475/222061435-c5e1aaf0-e879-4948-9dbf-88b6f9958d1a.png">

请问使用的 Paddle 版本是？麻烦提供一些更详细的环境信息方便定位问题～

**Comment by which520** [comment]
paddle的版本2.3.2和2.4.2和2.4.1都是试过了，然后python环境是直接使用主机上安装的python3.9 没有使用conda虚拟环境

**Comment by SigureMo** [comment]
> paddle的版本2.3.2和2.4.2和2.4.1都是试过了，然后python环境是直接使用主机上安装的python3.9 没有使用conda虚拟环境

我这边测试了下自带的 python3.9（`/usr/bin/python3`），同样是可以跑通的

<img width="846" alt="image" src="https://user-images.githubusercontent.com/38436475/222128915-38701f8e-19bb-44a2-a0aa-0b07e97bc026.png">

**Comment by which520** [comment]
应该是我环境的问题，我使用/usr/local/bin/python3就没问题了，使用/usr/bin/python3就会有问题
